dlite_entities_service/cli/main.py

```diff
@@ -0,0 +1,397 @@
+"""Typer CLI for doing SOFT entities service stuff."""
+from __future__ import annotations
+
+import json
+import os
+import re
+import sys
+from pathlib import Path
+from typing import TYPE_CHECKING, Optional
+
+if sys.version_info >= (3, 11):
+    from enum import StrEnum
+else:
+    from enum import Enum
+
+    class StrEnum(str, Enum):
+        """Enum with string values."""
+
+
+try:
+    import typer
+except ImportError as exc:  # pragma: no cover
+    from dlite_entities_service.cli._utils.generics import EXC_MSG_INSTALL_PACKAGE
+
+    raise ImportError(EXC_MSG_INSTALL_PACKAGE) from exc
+
+
+import yaml
+from dotenv import dotenv_values
+from pydantic import AnyHttpUrl
+
+from dlite_entities_service.cli._utils.generics import (
+    ERROR_CONSOLE,
+    pretty_compare_dicts,
+    print,
+)
+from dlite_entities_service.cli._utils.global_settings import CONTEXT, global_options
+from dlite_entities_service.cli.config import APP as config_APP
+from dlite_entities_service.models import (
+    URI_REGEX,
+    get_updated_version,
+    get_uri,
+    get_version,
+    soft_entity,
+)
+from dlite_entities_service.service.exceptions import BackendError
+
+if TYPE_CHECKING:  # pragma: no cover
+    from typing import Any
+
+    from pymongo.collection import Collection
+
+
+class EntityFileFormats(StrEnum):
+    """Supported entity file formats."""
+
+    JSON = "json"
+    YAML = "yaml"
+    YML = "yml"
+
+
+# Type Aliases
+OptionalListEntityFileFormats = Optional[list[EntityFileFormats]]
+OptionalListStr = Optional[list[str]]
+OptionalListPath = Optional[list[Path]]
+OptionalStr = Optional[str]
+
+
+APP = typer.Typer(
+    name="entities-service",
+    help="SOFT entities service utility CLI",
```

Comment:
I actually think I will be renaming this whole thing to simply remove `DLite` and `SOFT` wherever it's not applicable. So this would become `Entities service utility CLI` instead. For now I will keep SOFT here, since DLite is simply an implementation of SOFT-style data models.